crates/jetstreamer-source/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+        let token = cancellation_token.clone();
+
+        tokio::spawn(async move {
+            if let Err(e) = Self::stream_loop(config, filters, tx.clone(), token).await {
+                error!(error = %e, "Jetstream streaming failed");
+                let _ = tx
+                    .send(Err(yellowstone_grpc_proto::tonic::Status::internal(
+                        e.to_string(),
+                    )))
+                    .await;
+            }
+        });
+
+        Ok(())
+    }
+}
+
+impl JetstreamSource {
+    async fn stream_loop(
+        config: JetstreamSourceConfig,
+        _filters: Filters,
+        tx: Sender<Result<SubscribeUpdate, yellowstone_grpc_proto::tonic::Status>>,
+        cancellation_token: CancellationToken,
+    ) -> Result<(), Error> {
+        debug!("Entering stream_loop method");
+
+        let (start_slot, end_slot) = config.range.to_slot_range().map_err(|e| {
+            Error::SlotRangeResolution(format!("Failed to resolve slot range: {}", e))
+        })?;
+        info!(
+            start_slot,
+            end_slot, "Starting Old Faithful data streaming from {}", config.archive_url
+        );
+
+        let reorder_buffer = Arc::new(RwLock::new(ReorderBuffer::new(
+            start_slot,
+            config.reorder_buffer_size,
+            config.slot_timeout_secs,
+        )));
+
+        let buffer_clone = reorder_buffer.clone();
+        let tx_clone = tx.clone();
+        let token_clone = cancellation_token.clone();
+        tokio::spawn(async move {
+            Self::timeout_checker_loop(buffer_clone, tx_clone, token_clone).await;
+        });
+
+        let handler = Arc::new(VixenStreamHandler::new(tx.clone(), reorder_buffer.clone()));
+
+        std::env::set_var("JETSTREAMER_NETWORK", config.network.clone());
+        std::env::set_var(
+            "JETSTREAMER_COMPACT_INDEX_BASE_URL",
+            config.compact_index_base_url.clone(),
+        );
+        std::env::set_var(
+            "JETSTREAMER_NETWORK_CAPACITY_MB",
+            config.network_capacity_mb.to_string(),
+        );
+
+        let handler_clone = handler.clone();
+        let on_block = Some(move |_thread_id: usize, block: BlockData| {
+            let handler = handler_clone.clone();
+            async move {
+                handler
+                    .process_block(block)
+                    .await
+                    .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send>)
+            }
+            .boxed()
+        });
+
+        let handler_clone = handler.clone();
+        let on_tx = Some(move |_thread_id: usize, tx: TransactionData| {
+            let handler = handler_clone.clone();
+            async move {
+                handler
+                    .process_transaction(tx)
+                    .await
+                    .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send>)
+            }
+            .boxed()
+        });
```

Comment:
```suggestion
        let handler_on_tx = handler.clone();
        let on_tx = Some(move |_thread_id: usize, tx: TransactionData| {
            let handler_callback = handler_on_tx.clone();
            async move {
                handler
                    .process_transaction(tx)
                    .await
                    .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send>)
            }
            .boxed()
        });
```

A bit more expressive names for the handler clones.